lib/client/doc.go

```diff
@@ -14,5 +14,93 @@
 
 /*
 Package client implements the northbound client used to manage Calico configuration.
+
+This client is the main entry point for applications that are managing or querying
+Calico configuration.
+
+This client provides a typed interface for managing different resource types.  The
+definitions for each resource type are defined in the following package:
+	github.com/tigera/libcalico-go/lib/api
+
+The client has a number of methods that return interfaces for managing:
+	-  Tier resources
+	-  Policy resources
+	-  IP Pool resources
+	-  Host endpoint resources
+	-  Workload resources
+	-  Profile resources
+	-  IP Address Management
+
+See interface definitions for details about the set of management commands for each
+resource type.
+
+The resource management interfaces have a common set of commands to create, delete,
+update and retrieve resource instances.  For example, an application using this
+client to manage tier resources would create an instance of this client, create a
+new Tiers interface and call the appropriate methods on that interface.  For example:
```

Comment:
godoc has a [way of doing examples](https://godoc.org/github.com/fluhus/godoc-tricks#Examples) that turns them into UTs so they can't fail.  Might be worth looking into.